src/components/item/RelatedUnits.vue

```diff
@@ -0,0 +1,437 @@
+
+      const isChoosable = isSavedOnServer && !isItself && !isAlreadyRelated;
+
+      return isListChoosingOnlyToShowItems ? isSavedOnServer : isChoosable;
+    },
+    updateItemField({ field, stringsArray, objectsArray }) {  
+      this.updateItemFieldLocally({ field, value: stringsArray });
+      this.updateItemRelatedUnitsLocally({ field, value: objectsArray });
+
+      if (this.edittingItemObj.title || this.edittingItemObj.details) {
+        this[this.edittingItemObj.id 
+          ? '_updateItemOnServer' 
+          : '_addItemOnServer']({ 
+          item: this.edittingItemObj, 
+          cancelToken: null,
+        });
+      }
+    },
+    addRelatedItem() {
+      this.updateItemField({
+        field: 'relatedItems',
+        stringsArray: [
+          ...(this.edittingItemObj.relatedItems || []),
+          this.chosenItemId,
+        ],
+        objectsArray: [
+          ...(this.currentItemObj.relatedItems || []),
+          this.itemsFromPossibleRelatedList[this.chosenItemIndex],
+        ],
+      });
+
+      this.chosenItemId = null;
+    },
+    addRelatedList() {
+      this.updateItemField({
+        field: 'relatedLists',
+        stringsArray: [
+          ...(this.edittingItemObj.relatedLists || []),
+          this.chosenItemId,
+        ],
+        objectsArray: [
+          ...(this.currentItemObj.relatedLists || []),
+          this.lists[this.chosenListIndex],
+        ],
+      });
+
+      this.chosenListId = null;
+    },
+    deleteRelatedItem(id) {
+      const stringsArray = this.getArrayWithoutDeletedUnit({
+        initialArray: this.edittingItemObj.relatedItems,
+        unitIdToDelete: id,
+      });
+      const objectsArray = this.getArrayWithoutDeletedUnit({
+        initialArray: this.currentItemObj.relatedItems,
+        unitIdToDelete: id,
+      });
+
+      this.updateItemField({
+        field: 'relatedItems',
+        stringsArray,
+        objectsArray,
+      });
+    },
+    deleteRelatedList(id) {
+      const stringsArray = this.getArrayWithoutDeletedUnit({
+        initialArray: this.edittingItemObj.relatedLists,
+        unitIdToDelete: id,
+      });
+      const objectsArray = this.getArrayWithoutDeletedUnit({
+        initialArray: this.currentItemObj.relatedLists,
+        unitIdToDelete: id,
+      });
+
+      this.updateItemField({
+        field: 'relatedLists',
+        stringsArray,
+        objectsArray,
+      });
+    },
+    getArrayWithoutDeletedUnit({ initialArray, unitIdToDelete }) {
+      const arrayCopy = [...initialArray];
+      const indexToRemove = initialArray.findIndex(unit => {
+        return unitIdToDelete === (initialArray[0] instanceof Object ? unit.id : unit);
+      });
+
+      arrayCopy.splice(indexToRemove, 1);
+
+      return arrayCopy.length ? arrayCopy : null;
+    },
+  },
+};
+</script>
+
+<template>
+  <div class="related-units">
+    <div v-if="currentItemObj?.relatedItems?.length">
+      <h1>related items:</h1>
```

Comment:
it's better be classed and explicitly add a size
or make a reusable component for title where this title style would be explicitly declared